app/javascript/packs/application.js

```diff
@@ -0,0 +1,13 @@
+// This file is automatically compiled by Webpack, along with any other files
```

Comment:
1. Whats this file about?
2. How does this work?

Reply:
Similarly to the `application.css`, it's a manifest file that tells Sprockets which JS files we want to include to compile.
---
[Question]
1. Rails mentions that by serving one file rather than many, the load time of pages can be greatly reduced because the browser makes fewer requests. But doesn't that mean that the compiled file becomes much larger, requiring longer download times? (It is common for JS projects the bundle into smaller chunks, commonly by page routes to reduce file sizes from my knowledge.)

Reply:
> But doesn't that mean that the compiled file becomes much larger, requiring longer download times?

You are absolutely correct. That is why the frontend is supported with webpack here to support this breaking down into smaller chunks.